e2e/volume_ops_test.go

```diff
@@ -148,8 +151,51 @@ func testVolumeStop(t *testing.T) {
 
 func testVolumeList(t *testing.T) {
 	r := require.New(t)
+	filterParams := map[string]string{
+		"key":   "owner",
+		"value": "gd2test",
+	}
+	volumes, err := client.Volumes("", filterParams)
+	r.Nil(err)
+	r.Len(volumes, 1)
+
+	filterParams = map[string]string{
+		"key": "owner",
+	}
+	volumes, err = client.Volumes("", filterParams)
+	r.Nil(err)
+	r.Len(volumes, 1)
+
```

Comment:
Write some comment here to separate match vs no-match cases.

* key match
* value match
* key and value match

You can create an array of `map[string]string` values and loop over the matching path.

```go
for _, filter := range matchingQueries {
    volumes, err = client.Volumes("", filter)
    r.Nil(err)
    r.Len(volumes, 1)
}
```

Do the same for non-matching cases:

```go
for _, filter := range NonMatchingQueries {
    volumes, err = client.Volumes("", filter)
    r.Nil(err)
    r.Len(volumes, 0)
}
```